prow/cmd/horologium/main.go

```diff
@@ -58,20 +63,45 @@ type kubeClient interface {
 	CreateProwJob(kube.ProwJob) (kube.ProwJob, error)
 }
 
-func sync(kc kubeClient, cfg *config.Config, now time.Time) error {
+type cronClient interface {
+	SyncConfig(cfg *config.Config) error
+	QueuedJobs() []string
+}
+
+func sync(kc kubeClient, cfg *config.Config, cr cronClient, now time.Time) error {
 	jobs, err := kc.ListProwJobs(kube.EmptySelector)
 	if err != nil {
 		return fmt.Errorf("error listing prow jobs: %v", err)
 	}
 	latestJobs := pjutil.GetLatestPeriodics(jobs)
 
+	// TODO(krzyzacy): retire the interval check, migrate everything to use cron
+	if err := cr.SyncConfig(cfg); err != nil {
+		logrus.WithError(err).Error("Error syncing cron jobs.")
+	}
+
+	cronTriggers := map[string]bool{}
+	for _, job := range cr.QueuedJobs() {
+		cronTriggers[job] = true
+	}
+
 	for _, p := range cfg.Periodics {
 		j, ok := latestJobs[p.Name]
-		if !ok || (j.Complete() && now.Sub(j.Status.StartTime) > p.GetInterval()) {
-			if _, err := kc.CreateProwJob(pjutil.NewProwJob(pjutil.PeriodicSpec(p), p.Labels)); err != nil {
-				return fmt.Errorf("error creating prow job: %v", err)
+
+		if p.Cron == "" {
+			if !ok || (j.Complete() && now.Sub(j.Status.StartTime) > p.GetInterval()) {
+				if _, err := kc.CreateProwJob(pjutil.NewProwJob(pjutil.PeriodicSpec(p), p.Labels)); err != nil {
+					return fmt.Errorf("error creating prow job: %v", err)
```

Comment:
maybe we should collect these errors and keep trying to create the other jobs?

Reply:
done